pkg/storage/tsdb/users_scanner.go

```diff
@@ -46,6 +47,10 @@ func (s *UsersScanner) ScanUsers(ctx context.Context) (users, markedForDeletion
 		return nil, nil, err
 	}
 
+	rand.Shuffle(len(users), func(i, j int) {
```

Comment:
I know we've changed course on this a time or two, but `ScanUsers` is a general purpose accessor function used by many services, whereas this shuffle behavior is particular to the needs of compactor. I actually think we can move the shuffle to `runCleanup` after we call `refreshOwnedUsers`, with a brief handrail comment noting why we'd do such a thing.